File: python/1874.py
import sys
input = sys.stdin.readline

n = int(input())

stackA = [0]
stackB = [i for i in range(1, n+1)]
stackB.reverse()
pushAndPop = []
flag = True

for _ in range(n):
    a = int(input())
    while(True):
        if stackA[-1] == a:
            stackA.pop()
            pushAndPop.append('-')
            break
        else:
            if len(stackB) == 0 and len(pushAndPop) != n*2:
                flag = False
                break
            stackA.append(stackB.pop())
            pushAndPop.append('+')

if flag == True:
    for i in pushAndPop:
        print(i)
else:
    print('NO')

File: c++/1874.cpp
#include <iostream>
#include <string>
#include <stack>

using namespace std;

int main(){
  int n;
  stack<int> s;
  string pm = "";
  int k=0;
  cin >> n;

  for(int i=0; i<n; i++){
    int a;
    cin >> a;
    if(a > k){
      while(a > k){
        s.push(++k);
        pm += '+';
      }
      s.pop();
      pm += '-';
    }
    else{
      bool flag = true;
      if(a != s.top()){
        flag = false;
      }
      s.pop();
      pm += '-';
      if(!flag){
        cout << "NO";
        return 0;
      }
    }
  }

  for(char c : pm){
    cout << c << '\n';
  }
  
  return 0;
}